### PR TITLE
added DIYRuZ Zintercom support

### PR DESCRIPTION
**Pull request by xyzroe**

### CODE REVIEW

**Comment by xyzroe** [comment]
Could anyone explain the meaning of this error?
It works as it should, z2m does not give errors, but the tests do not pass.

File: converters/fromZigbee.js [hunk 1]
```diff
@@ -5239,6 +5239,38 @@ const converters = {
             return result;
         },
     },
+    diyruz_zintercom_config: {
+        cluster: 'closuresDoorLock',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const result = {};
+            if (msg.data.hasOwnProperty(0x0050)) {
+                result.state = ['Idle', 'Ring', 'Talk', 'Open', 'Drop'][msg.data[0x0050]];
```
Comment (Koenkk) [inline converters/fromZigbee.js:5248]: Please make these values lowercase

File: converters/fromZigbee.js [hunk 2]
```diff
@@ -5239,6 +5239,38 @@ const converters = {
...
+            }
+            if (msg.data.hasOwnProperty(0x0051)) {
+                result.mode = ['Never', 'Once', 'Always', 'Drop'][msg.data[0x0051]];
```
Comment (Koenkk) [inline converters/fromZigbee.js:5251]: Please make these values lowercase

**Comment by Koenkk** [comment]
Note to self: after this merge https://github.com/Koenkk/zigbee2mqtt.io/pull/830

File: converters/toZigbee.js [hunk 3]
```diff
@@ -4136,6 +4136,42 @@ const converters = {
             await entity.read(payloads[key][0], [payloads[key][1]]);
         },
     },
+    diyruz_zintercom_config: {
+        key: ['mode', 'sound', 'time_ring', 'time_talk', 'time_open', 'time_bell', 'time_report'],
+        convertSet: async (entity, key, rawValue, meta) => {
+            const lookup = {'OFF': 0x00, 'ON': 0x01};
+            const modeOpenLookup = {'Never': '0', 'Once': '1', 'Always': '2', 'Drop': '3'};
```
Comment (Koenkk) [inline converters/toZigbee.js:4143]: This also has to be made lower case.

File: devices/diyruz.js [hunk 4]
```diff
@@ -256,4 +256,42 @@ module.exports = [
...
+            const payload1 = [
+                {attribute: 'batteryPercentageRemaining', minimumReportInterval: 0, maximumReportInterval: 3600, reportableChange: 0},
+                {attribute: 'batteryVoltage', minimumReportInterval: 0, maximumReportInterval: 3600, reportableChange: 0}];
+            await firstEndpoint.configureReporting('genPowerCfg', payload1);
+            const payload2 = [{attribute: {ID: 0x0050, type: 0x30},
+                minimumReportInterval: 0, maximumReportInterval: 3600, reportableChange: 0}];
+            await firstEndpoint.configureReporting('closuresDoorLock', payload2);
+        },
+        exposes: [
+            exposes.enum('state', ea.STATE, ['Idle', 'Ring', 'Talk', 'Open', 'Drop'])
```
Comment (Koenkk) [inline devices/diyruz.js:278]: This also has to be made lower case.

File: devices/diyruz.js [hunk 5]
```diff
@@ -256,4 +256,42 @@ module.exports = [
...
+                .withDescription('Current state'),
+            exposes.enum('mode', ea.ALL, ['Never', 'Once', 'Always', 'Drop'])
```
Comment (Koenkk) [inline devices/diyruz.js:280]: This also has to be made lower case.

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**